Application/EdFi.Ods.Tests/EdFi.Ods.Common/Database/PrototypeTokenReplacementOverrideDatabaseConnectionStringProviderTests.cs

```diff
@@ -137,4 +145,61 @@ public void Should_properly_modify_connection_string_template_with_varying_datab
             _actualConnectionString3.ShouldContain("=EdFi_OneDatabase;");
         }
     }
+
+    public class When_building_connection_string_based_on_a_prototype_from_the_connectionStrings_config_section_with_different_server_names
+        : TestFixtureBase
+    {
+        private string _actualConnectionStringWithPrefix;
+        private string _actualConnectionStringWithoutTemplate;
+
+        private IDatabaseNameReplacementTokenProvider _databaseNameReplacementTokenProvider;
+        private IConfigConnectionStringsProvider _configConnectionStringsProvider;
+        private IDbConnectionStringBuilderAdapterFactory _dbConnectionStringBuilderAdapterFactory;
+        private IDatabaseServerNameProvider _databaseServerNameProvider;
+
+        protected override void Arrange()
+        {
+            _databaseNameReplacementTokenProvider = A.Fake<IDatabaseNameReplacementTokenProvider>();
+
+            _databaseServerNameProvider = A.Fake<IDatabaseServerNameProvider>();
+
+            A.CallTo(() => _databaseServerNameProvider.GetDatabaseServerName()).Returns("OneDatabase");
+
+            _configConnectionStringsProvider = A.Fake<IConfigConnectionStringsProvider>();
+
+            A.CallTo(() => _configConnectionStringsProvider.Count).Returns(1);
+
+            A.CallTo(() => _configConnectionStringsProvider.GetConnectionString(A<string>.Ignored))
+                .Returns("Server=SomeServer; Database=SomeDatabase; UID=SomeUser; Password=SomePassword")
+                .Once();
+
+            A.CallTo(() => _configConnectionStringsProvider.GetConnectionString(A<string>.Ignored))
+                .Returns("Server=prefix_{0}_suffix; Database=SomeDatabase; UID=SomeUser; Password=SomePassword")
+                .Once();
+
+            _dbConnectionStringBuilderAdapterFactory = A.Fake<IDbConnectionStringBuilderAdapterFactory>();
+
+            A.CallTo(() => _dbConnectionStringBuilderAdapterFactory.Get()).Returns(new SqlConnectionStringBuilderAdapter());
+        }
+
+        protected override void Act()
+        {
+            var provider = new PrototypeTokenReplacementConnectionStringProvider(
+                "SomeConnectionStringName",
+                _databaseNameReplacementTokenProvider,
+                _configConnectionStringsProvider,
+                _dbConnectionStringBuilderAdapterFactory,
+                _databaseServerNameProvider);
+
+            _actualConnectionStringWithPrefix = provider.GetConnectionString();
+            _actualConnectionStringWithoutTemplate = provider.GetConnectionString();
+        }
+
+        [Assert]
+        public void Should_properly_modify_connection_string_template_with_varying_server_names()
+        {
+            _actualConnectionStringWithPrefix.ShouldContain("=prefix_OneDatabase_suffix;");
+            _actualConnectionStringWithoutTemplate.ShouldContain("=SomeServer;");
```

Comment:
The way these checks were done originally for DatabaseName were a bit sketchy, but particularly now that we're working with multiple potential token replacements, we should enhance these asserts to explicitly check for "Database=SomeServer" rather than just "=SomeServer".  We run the risk of a false positive on this test doing the checks this way in case some other portion of the connection string happens to match up.

Reply:
Updated checks to differentiate database vs server.